### PR TITLE
Offline UX

### PR DESCRIPTION
**Pull request by JurreBrandsenInfoSupport**
@nicojs This is the improvement of #16 

---
In this pull request, we introduce a feature that detects when the user is offline using the navigator.onLine listener. When the user goes offline, we display a persistent toast message to inform them of their offline status while ensuring they can still interact with the survey seamlessly. Upon detecting that the user is back online, the current state is promptly sent to the database for synchronization.

---
We have the option to extend this feature by integrating an interval of `n` seconds, where the current state is regularly transmitted to the database for ongoing synchronization. However, it's currently not imperative, as we already send the state to the database every time the user marks an answer. If the stress test ( #18 ) reveals an excessive load on the database due to these frequent calls, we can implement this adjustment accordingly.

### CODE REVIEW

File: src/components/mobile/survey-questions.tsx [hunk 1]
```diff
@@ -82,6 +91,83 @@ export function MobileSurveyQuestionnaire({
...
+          toast({
+            title: "Failed to resend responses",
+            description:
+              "An error occurred while attempting to resend responses.",
+            variant: "destructive",
+          });
+        }
+        PreviouslyOffline = false;
+      }
+    };
```
Comment (nicojs) [inline src/components/mobile/survey-questions.tsx:115]: I would expect a `useOnlineStatus` effect, a simple boolean state. Then, I would the toaster to be handled higher up on the page. Here I would only expect the answer buffering logic

File: src/components/mobile/survey-questions.tsx [hunk 2]
```diff
@@ -82,6 +91,83 @@ export function MobileSurveyQuestionnaire({
...
+            title: "Failed to resend responses",
+            description:
+              "An error occurred while attempting to resend responses.",
+            variant: "destructive",
+          });
+        });
+      });
+      window.removeEventListener("offline", handleOffline);
+    };
+  }, [responses, session, submitResponse]);
```
Comment (nicojs) [inline src/components/mobile/survey-questions.tsx:169]: I think this effect runs more often than you realize. `submitResponse` seems to be a new object every time, so this effect runs effectively on every cycle.

File: src/components/mobile/survey-questions.tsx [hunk 3]
```diff
@@ -82,6 +91,83 @@ export function MobileSurveyQuestionnaire({
     },
   });
 
+  useEffect(() => {
```
Comment (nicojs) [inline src/components/mobile/survey-questions.tsx:94]: In general, this effect runs far too long. In general, try to not use `useEffect` directly from a component, instead move it to helper methods.

File: src/components/mobile/survey-questions.tsx [hunk 4]
```diff
@@ -122,9 +208,10 @@ export function MobileSurveyQuestionnaire({
     <div>
       <Form {...form}>
         <form
-          onSubmit={form.handleSubmit((data) => {
+          onSubmit={form.handleSubmit(async () => {
```
Comment (nicojs) [inline src/components/mobile/survey-questions.tsx:211]: Why have 2 components that do the same state management? The mobile component looks identical.

**Review by nicojs — CHANGES_REQUESTED** [review]
I didn't have time to look through all of it, but let's have a look tomorrow.

File: src/components/mobile/survey-questions.tsx (hunk 4 above)
Comment (JurreBrandsenInfoSupport) [reply src/components/mobile/survey-questions.tsx]: Solved in #46

File: src/components/mobile/survey-questions.tsx (hunk 1 above)
Comment (JurreBrandsenInfoSupport) [reply src/components/mobile/survey-questions.tsx]: Created #47

File: src/components/mobile/survey-questions.tsx (hunk 2 above)
Comment (JurreBrandsenInfoSupport) [reply src/components/mobile/survey-questions.tsx]: Created #47

**Comment by JurreBrandsenInfoSupport** [comment]
Closed. Accepted changes are merged in #46. Open Issues #47

**Pull request closed without merging**